nginx-controller/nginx/nginx.go

```diff
@@ -30,8 +30,9 @@ type NginxController struct {
 
 // IngressNginxConfig describes an NGINX configuration
 type IngressNginxConfig struct {
-	Upstreams []Upstream
-	Servers   []Server
+	Upstreams     []Upstream
+	Servers       []Server
+	EndpointsInfo map[string]EndpointInfo
```

Comment:
I don't think `EndpointsInfo` should be a part of the `IngressNginxConfig`. It is better to have a special type for a map. Something like this:
```go
type Map struct {
  Source string
  Variable string
  Params map[string]string
}